### PR TITLE
Update

### PR DESCRIPTION
**Pull request by trapacska**

### CODE REVIEW

File: bitrise.yml [hunk 1]
```diff
@@ -51,29 +43,31 @@ workflows:
...
+        - project_path: ./macos-sample-app.xcodeproj
+        - scheme: macos-sample-app
+        - export_method: app-store
+        - output_tool: xcodebuild
+        - custom_export_options_plist_content: $EXPORT_PLIST
 
   test:
     steps:
     - path::./:
         title: Step Test - Export Options
```
Comment (godrei) [inline bitrise.yml]: Step Test - App Store, also update following test step titles (- path::./:)

File: main.go [hunk 2]
```diff
@@ -110,8 +118,54 @@ func (configs ConfigsModel) validate() error {
 	return nil
 }
 
+func printCertificateInfo(info certificateutil.CertificateInfoModel) {
```
Comment (godrei) [inline main.go]: unused?

File: main.go [hunk 3]
```diff
@@ -110,8 +118,54 @@ func (configs ConfigsModel) validate() error {
...
+	log.Printf("serial: %s", info.Serial)
+	log.Printf("team: %s (%s)", info.TeamName, info.TeamID)
+	log.Printf("expire: %s", info.EndDate)
+
+	if err := info.CheckValidity(); err != nil {
+		log.Errorf("[X] %s", err)
+	}
+}
+
+func printProfileInfo(info profileutil.ProvisioningProfileInfoModel, installedCertificates []certificateutil.CertificateInfoModel) {
```
Comment (godrei) [inline main.go]: unused?

File: main.go [hunk 4]
```diff
@@ -140,178 +194,327 @@ func main() {
...
+	}
 
-		if exportingApp {
-			legacyExportCmd.SetExportPath(appPath)
-		} else {
-			legacyExportCmd.SetExportPath(pkgPath)
-		}
+	// legacy export
+	{
+		if configs.UseLegacyExport == "yes" {
```
Comment (godrei) [inline main.go]: fail if legacy options are set and xcode major version >= 9 like: https://github.com/bitrise-io/steps-xcode-archive/pull/95/files#diff-7ddfb3e035b42cd70649cc33393fe32cR472

File: main.go [hunk 5]
```diff
@@ -140,178 +194,327 @@ func main() {
...
-			fail("Export failed, error: %s", err)
-		}
+				provisioningProfileName = configs.LegacyExportProvisioningProfileName
+			} else {
+				log.Printf("Using embedded provisioning profile")
 
-		if exportingApp {
-			if err := utils.ExportOutputFile(appPath, appPath, bitriseAppPathEnvKey); err != nil {
-				fail("Failed to export %s, error: %s", bitriseAppPathEnvKey, err)
+				provisioningProfileName = archive.Application.ProvisioningProfile.Name
```
Comment (godrei) [inline main.go:229]: archive.Application.ProvisioningProfile mac os archive's profile is mostly nil, check it and fail if no profile embedded!

File: main.go [hunk 6]
```diff
@@ -140,178 +194,327 @@ func main() {
...
+			installedCertificates, err := certificateutil.InstalledCodesigningCertificateInfos()
+			if err != nil {
+				fail("Failed to get installed certificates, error: %s", err)
 			}
-		} else {
-			log.Detail("Generating export options")
 
-			method, err := exportoptions.ParseMethod(configs.ExportMethod)
+			log.Debugf("\n")
+			log.Debugf("Installed certificates:")
```
Comment (godrei) [inline main.go]: log.Debugf("Installed certificates:") -> log.Debugf("\nInstalled certificates:")

File: Gopkg.toml [hunk 7]
```diff
@@ -0,0 +1,30 @@
+
+# Gopkg.toml example
```
Comment (kdobmayer) [inline Gopkg.toml]: Remove comments

File: Gopkg.toml [hunk 8]
```diff
@@ -0,0 +1,30 @@
...
+#   name = "github.com/user/project2"
+#   branch = "dev"
+#   source = "github.com/myfork/project2"
+#
+# [[override]]
+#  name = "github.com/x/y"
+#  version = "2.4.0"
+
+
+[[constraint]]
```
Comment (kdobmayer) [inline Gopkg.toml]: `name` first, then `branch`

File: main.go [hunk 9]
```diff
@@ -140,178 +148,331 @@ func main() {
...
-
 	appPath := filepath.Join(configs.DeployDir, archiveName+".app")
 	pkgPath := filepath.Join(configs.DeployDir, archiveName+".pkg")
 	exportOptionsPath := filepath.Join(configs.DeployDir, "export_options.plist")
-
 	ideDistributionLogsZipPath := filepath.Join(configs.DeployDir, "xcodebuild.xcdistributionlogs.zip")
 
+	xcodebuildVersion, err := utility.GetXcodeVersion()
+	if err != nil {
+		fail("Failed to determin xcode version, error: %s", err)
```
Comment (kdobmayer) [inline main.go]: typo: determin**e**

File: main.go [hunk 10]
```diff
@@ -140,178 +148,331 @@ func main() {
...
+			fmt.Println()
+			log.Printf("Target Bundle ID - Entitlements map")
+			idx := -1
+			for bundleID, entitlements := range bundleIDEntitlementsMap {
+				idx++
+				entitlementKeys := []string{}
+				for key := range entitlements {
+					entitlementKeys = append(entitlementKeys, key)
+				}
+				log.Printf("%s: [%s]", bundleID, strings.Join(entitlementKeys, " "))
```
Comment (kdobmayer) [inline main.go]:
```go
log.Printf("%s: %s", bundleID, entitlementKeys)
```

**Pull request merged**